core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataDeleteIndexService.java

```diff
@@ -49,6 +50,17 @@
  */
 public class MetaDataDeleteIndexService extends AbstractComponent {
 
+    /**
+     * Setting for the maximum tombstones allowed in the cluster state;
+     * prevents the cluster state size from exploding too large, but it opens the
+     * very unlikely risk that if there are greater than MAX_TOMBSTONES index
+     * deletions while a node was offline, when it comes back online, it will have
+     * missed index deletions that it may need to process.
+     */
+    public static final Setting<Integer> SETTING_MAX_TOMBSTONES = Setting.intSetting("cluster.indices.tombstones.size",
+                                                                                     500, // the default maximum number of tombstones
+                                                                                     Setting.Property.NodeScope);
```

Comment:
Just discussed with @dakrone, there is no cluster scope, `Setting.Property.NodeScope` is used for cluster level.